messaging/src/main/java/org/axonframework/queryhandling/SimpleQueryBus.java

```diff
@@ -247,33 +251,24 @@ public static Builder builder() {
             throw new IllegalArgumentException("There is already a subscription with the given message identifier");
         }
 
-        MonoWrapper<QueryResponseMessage<I>> initialResult = getInitialResultMono(query);
+        Mono<QueryResponseMessage<I>> initialResult = Mono
+                .fromFuture(query(query))
```

Comment:
I think we should add a `Mono.defer()` call here to ensure the `CompletableFuture` is only created when the Mono is subscribed to.